vmm/src/vm_config.rs

```diff
@@ -1056,4 +1065,45 @@ impl VmConfig {
             self.cpus.max_vcpus
         }
     }
+
+    pub fn consume_fds(&mut self, fds: Vec<i32>) -> Result<(), PayloadConfigError> {
```

Comment:
We have a complex mechanism so we should add proper documentation to the underlying mechanism. The mechanism is complex enough to write a few paragraphs about it:

- explain the rather complex overall mechanism
- explain constraints for developers for their public APIs
- explain how FDs are associated with devices of a VM
- what else might be interesting

Or in other words. We should not worsen the situation that Cloud Hypervisor code is undocumented. New functionality especially for complex stuff should have documentation. I don't say must as I'm not a maintainer, yet I'm a major contributor :)